File: observer/typescript/observer-user.interface.ts
import {ProductUpdateMessage} from "./product-update-message";

// Değişimden etkilenecek, izleyecek olan sınıfların uyguladığı arayüzdür.
// UML diyagramındaki Observer yapısına denk gelmektedir.
export interface IObserverUser {
  sendNotification(message: ProductUpdateMessage): void;
}

File: observer/go/observer-user-interface.go
package main

// ObserverUser Değişimden etkilenecek, izleyecek olan sınıfların uyguladığı arayüzdür.
// UML diyagramındaki Observer yapısına denk gelmektedir.
type ObserverUser interface {
	SendNotification(msg ProductUpdateMessage)
}
